### ISSUE TITLE
In below NER traininig code , what are the parameters that I can tweak other than dropout rate?

### ISSUE THREAD

**Issue by prashant334** [explosion/spaCy#1848]
* **Python version:** 2.7.6
* **Platform:** Linux-3.16.0-77-generic-x86_64-with-Ubuntu-14.04-trusty
* **spaCy version:** 2.0.0a17
* **Models:** en, en_core_web_sm, xx_ent_wiki_sm

@honnibal @ines 

CODE:
```python
def train_ner(nlp, train_data, output_dir):
    random.seed(0)
    optimizer = nlp.begin_training(lambda: [])
    nlp.meta['name'] = 'CRIME_LOCATION'
    for itn in range(50):
        losses = {}
        for batch in minibatch(get_gold_parses(nlp.make_doc, train_data), size=3):
            docs, golds = zip(*batch)
            nlp.update(docs, golds, losses=losses, sgd=optimizer,
                       **drop=0.35**)
        print("under learning")
    if not output_dir:
        return
```

**Comment by ines** [comment]
Your code already uses all the available keyword arguments of `Language.update` – see here for the API docs: http://spacy.io/api/language#update

But you might also want to check out [`spacy train`](http://spacy.io/api/cli#train) and the [available hyperparameters](http://spacy.io/api/cli#train-hyperparams) that can be set as environment variables. You can see the Python implementation of the training command [here](https://github.com/explosion/spaCy/blob/master/spacy/cli/train.py). For more details on some of the tricks used here, like the compounding batch sizes and decaying dropout rate, see the [optimization tips and advice](https://spacy.io/usage/training#section-tips).

**Comment by prashant334** [comment]
@ines 

I want to change optimizer in the same program . Is there any way to do it directly in program or I need to do it using in command line.
